### PR TITLE
You can now buy an ERT from the communications console

### PR DESCRIPTION
**Pull request by Simplehorror**
# Document the changes in your pull request

You can now buy an ERT from the comms console. 
Red ERT = 100000 credits
Blue = 30000
Amber = 10000
Janitor = 20000
Holy = 50000

Admins are VERY RELUCTANT to spawn ERTs even when everything has gone to shit, so making it a code-based thing makes it more consistent which is more fun to the players, because you know when to expect it, and it makes it easier to balance since it's not admin-only anymore.

# Spriting

# Wiki Documentation

todo

# Changelog

:cl:  
tweak: You can now buy an ERT from the communications console
/:cl:

### CODE REVIEW

**Comment by MajManatee** [comment]
you realize that 100k is made in like...30 seconds?

**Comment by Simplehorror** [comment]
Limitations I want to add:

- You can only order after so many security officers have died/became antagonists
- You can only order an ERT once
- You can only order certain ERTs if there is station destruction > 20%
- You can only order after a fair amount of time

**Review by Moltijoe — CHANGES_REQUESTED** [review]
Should probably cap it at 1 ERT per shift

**Comment by Simplehorror** [comment]
> you realize that 100k is made in like...30 seconds?

blame jungleland

**Comment by Simplehorror** [comment]
yeah that's the plan

**Comment by MajManatee** [comment]
> > you realize that 100k is made in like...30 seconds?
> 
> blame jungleland

I make 3-400k EVERY SHIFT on lava land.

**Comment by Rdrazga** [comment]
Why not WC ert instead of regular, also where my peacekeepers at?

**Comment by Simplehorror** [comment]
> > > you realize that 100k is made in like...30 seconds?
> > 
> > 
> > blame jungleland
> 
> I make 3-400k EVERY SHIFT on lava land.

it takes you 1h45m to get 400k, which means this seems pretty balanced to me seeing as if you need a red ert you won't be making that much money

**Comment by Simplehorror** [comment]
> Why not WC ert instead of regular, also where my peacekeepers at?

this is WC ert, and peacekeepers looked sucky

**Comment by Rdrazga** [comment]
Also only allow it if red alert is on

**Comment by Simplehorror** [comment]
> Also only allow it if red alert is on

yeah fair point

**Comment by Rdrazga** [comment]
> > Why not WC ert instead of regular, also where my peacekeepers at?
> 
> this is WC ert, and peacekeepers looked sucky

Yeah they are sucky but are basically more police focused then amber, if you are missing a sec team being able to order 2-3 peacekeepers to replace them would be good

**Comment by Rdrazga** [comment]
Also would be very fun if emmaged comms console could order clowns

**Comment by Simplehorror** [comment]
i was thinking about it letting you order a lone op

**Comment by MajManatee** [comment]
> > > > you realize that 100k is made in like...30 seconds?
> > > 
> > > 
> > > blame jungleland
> > 
> > 
> > I make 3-400k EVERY SHIFT on lava land.
> 
> it takes you 1h45m to get 400k, which means this seems pretty balanced to me seeing as if you need a red ert you won't be making that much money

have you played cargo? ever?
I get to 400k by the 30 minute mark, pretty consistently

**Comment by Rdrazga** [comment]
> i was thinking about it letting you order a lone op

Maybe an extra traitor buddy with objectives to help you and no (or low) tc not a full on line op

**Comment by Simplehorror** [comment]
> > > > > you realize that 100k is made in like...30 seconds?
> > > > 
> > > > 
> > > > blame jungleland
> > > 
> > > 
> > > I make 3-400k EVERY SHIFT on lava land.
> > 
> > 
> > it takes you 1h45m to get 400k, which means this seems pretty balanced to me seeing as if you need a red ert you won't be making that much money
> 
> have you played cargo? ever? I get to 400k by the 30 minute mark, pretty consistently

i have, unless cargo got buffed in the last year or so i don't remember getting that much outside of stocks

**Comment by Rdrazga** [comment]
> > > > > > you realize that 100k is made in like...30 seconds?
> > > > > 
> > > > > 
> > > > > blame jungleland
> > > > 
> > > > 
> > > > I make 3-400k EVERY SHIFT on lava land.
> > > 
> > > 
> > > it takes you 1h45m to get 400k, which means this seems pretty balanced to me seeing as if you need a red ert you won't be making that much money
> > 
> > 
> > have you played cargo? ever? I get to 400k by the 30 minute mark, pretty consistently
> 
> i have, unless cargo got buffed in the last year or so i don't remember getting that much outside of stocks

Selling plasma, engineering rad collectors

**Comment by Simplehorror** [comment]
> > > > > > > you realize that 100k is made in like...30 seconds?
> > > > > > 
> > > > > > 
> > > > > > blame jungleland
> > > > > 
> > > > > 
> > > > > I make 3-400k EVERY SHIFT on lava land.
> > > > 
> > > > 
> > > > it takes you 1h45m to get 400k, which means this seems pretty balanced to me seeing as if you need a red ert you won't be making that much money
> > > 
> > > 
> > > have you played cargo? ever? I get to 400k by the 30 minute mark, pretty consistently
> > 
> > 
> > i have, unless cargo got buffed in the last year or so i don't remember getting that much outside of stocks
> 
> Selling plasma, engineering rad collectors

i sold all the mats and i never got close to 400k

**Comment by Rdrazga** [comment]
Depends on mining team but it is possible, for a normal cargo team though when you also take into account that they are likely buying stuff throughout a shift 100k or less is fairly common point for a station to be around. This does not include mass selling matts for emergencies or breaking into the sec budget

**Comment by Rdrazga** [comment]
If an atmos tech does set up a proper money rad collector though cargo will have multiple millions by shift end

**Comment by Simplehorror** [comment]
and you'll get one red ert congrats

**Comment by Sniblet** [comment]
.s CULT CONFIRMED
.e MONEY COLLECTORS NOW
.u SELL PLASMA
20 minutes pass, cult gets wiped in a fair and exciting battle involving tactical energy guns

**Comment by Simplehorror** [comment]
sounds like cargo was powercreeped while I was away if they can get 100k this easy

**Comment by Hopekz** [comment]
I think this PR can be good by only allowing the purchase of the weaker ERT's just to replenish security like peacekeepers then saving the rest for admins 

Also remember that the shutter on centcom for ERT's used to start  closed so if that still happens you'd have to find a solution for that so the bought ERT's can get down without admin intervention 

Ideally if I was making this PR I'd just allow peacekeepers for now because they are sec reinforcements and you said you wanted to make it to where people can call these after sec is dying. That way the station can only replenish their sec for a large-ish credit cost.

**Comment by Simplehorror** [comment]
peacekeepers are dogshit and all they're good for is stopping the clown being a menace, and will die to any intelligent antag

**Comment by Moltijoe** [comment]
> peacekeepers are dogshit and all they're good for is stopping the clown being a menace, and will die to any intelligent antag

PERFECT! so they'll fit right in with the rest of sec

**Comment by Hopekz** [comment]
> peacekeepers are dogshit 

I made peacekeepers so I know they're just basically reskinned security officers to replace a dead sec.

They have security officer access to re-arm in sec and arrest and if they can't re-arm they have a basic security officer loadout with disabler (wt with rubber rounds) stun baton (stun prod) and cuffs.

If you give them an objective that makes it clear they're replacing security like " You are sent to replenish the security department" they'll understand their role more since most people that play peacekeeper don't realize that they're just reskinned security officers with security officer access.

**Comment by Simplehorror** [comment]
right but nobody needs these. if all of security is wiped out you don't need more redshirts, you need something to kill the guys killing security. it's a solution without a problem. giving antags more fodder to kill isn't going to solve anything

**Comment by Hopekz** [comment]
> right but nobody needs these. if all of security is wiped out you don't need more redshirts, you need something to kill the guys killing security. it's a solution without a problem. giving antags more fodder to kill isn't going to solve anything

You can try your way but I have a feeling they won't merge it if you can spawn all the ERT's vs just the ones to replenish sec.

Remember it's okay for antagonists to win the round. The "solution" shouldn't be to actually provide a guaranteed dub for the station but rather to reintroduce more people into the round.

Either way I think this is a good idea for a PR game direction wise even if I don't 100% believe in the current execution of the idea.

**Comment by Simplehorror** [comment]
this isn't getting merged no matter what I do

**Comment by ChuChu42** [comment]
like this PR, on my way to spawn 10 red ERTs to replace entire station

**Review by Moltijoe — CHANGES_REQUESTED** [review]
Money based respawns, probably needs the balance council to look at this
Regardless, should cost more, have less ERT selection, and be limited to one per shift
Should also make sure that the ERT can actually get to the station once they spawn
not to mention the mission breifing is kind of a big part of ERTs

**Comment by RG4ORDR** [comment]
This both a stupid and interesting idea that I wanna see how it'd work out test merged.

**Comment by Marmio64** [comment]
Should only allow peacekeeper erts to be bought, with a per alive player limit on ert numbers (more ert the more dead people), and they should be relatively expensive. Money can often be no object and antags can be quickly swamped by powerful erts, so they should def be very weak. Also make sure admins can deny purchases.

**Comment by Simplehorror** [comment]
> Should only allow peacekeeper erts to be bought, with a per alive player limit on ert numbers (more ert the more dead people), and they should be relatively expensive. Money can often be no object and antags can be quickly swamped by powerful erts, so they should def be very weak. Also make sure admins can deny purchases.

that defeats the whole point of the pr

**Comment by boba247** [comment]
I think this could be good to try, currently, the station's only recourse to major emergencies is to call the shuttle. I do think Holy, Blue, and Red are too OP to start. We should start with Amber, peacekeeper, janitor, and clowns. Can revisit balance once the lesser-powered teams are tried out.

**Comment by ReddicusDragon** [comment]
If this isn't done, would you consider switching it to a draft please, SimpleHorror?

**Comment by Moltijoe** [comment]
> Limitations I want to add:
> 
> * You can only order after so many security officers have died/became antagonists
> * You can only order an ERT once
> * You can only order certain ERTs if there is station destruction > 20%
> * You can only order after a fair amount of time

make this a draft PR

File: code/datums/ert.dm [hunk 1]
```diff
@@ -19,6 +19,83 @@
...
+			ERTOperative.mind.assigned_role = ert_antag.name
+
+			// Equip uplink
+			var/obj/item/ntuplink/upl = new uplinktype(ERTOperative, ERTOperative.key)
+			if(istype(upl))
+				ERTOperative.equip_to_slot_or_del(upl, SLOT_IN_BACKPACK)
+				ert_team.uplink_type = uplinktype // Type path
+
+			//Logging and cleanup
+			//log_game("[key_name(ERTOperative)] has been selected as an [ert_antag.name]") | yogs - redundant
```
Comment (ReddicusDragon) [inline code/datums/ert.dm:80]: how is this redundant, or am I missing something here

Comment (Simplehorror) [reply code/datums/ert.dm:80]: came from the code i pasted

**Comment by Simplehorror** [comment]
idk how to draft a pr after you open it

**Comment by Moltijoe** [comment]
> idk how to draft a pr after you open it

image

**Comment by Bibby0110** [comment]
ERTs are specifically meant to be only spawned based on admin judgement

**Pull request closed without merging**